lib/models/message.dart

```diff
@@ -3,32 +3,27 @@ class Message {
   final MessageReceiverType receiverType;
   final String? teamId;
   final String? value;
-  final String? timestamp;
+  final String timestamp;
 
   Message(
       {required this.type,
       required this.receiverType,
       this.teamId,
       this.value,
-      this.timestamp});
+      required this.timestamp});
 
   Message.fromJson(Map<String, String> json)
       : type = MessageType.values.byName(json['type']!),
         receiverType = MessageReceiverType.values.byName(json['receiverType']!),
         teamId = json['teamId'],
         value = json['value'],
-        timestamp = json['timestamp'];
-
-  Message.fromJsonWithId(Map<String, String> json, String this.timestamp)
-      : type = MessageType.values.byName(json['type']!),
-        receiverType = MessageReceiverType.values.byName(json['receiverType']!),
-        teamId = json['teamId'],
-        value = json['value'];
+        timestamp = json['timestamp']!;
 
   Map<String, String> toJson() {
     Map<String, String> json = {
       'type': type.name,
-      'receiverType': receiverType.name
+      'receiverType': receiverType.name,
+      'timestamp': timestamp
```

Comment:
do we really need this? the timestamp is already a key

Reply:
You've added this in you pr. But I would argue that it's better to have it also inside payload, it's easier to get it

Reply:
yes but i was not sending this to db as an additional field in the object